src/role.ts

```diff
@@ -0,0 +1,317 @@
+   * @param grantees The resource(s) to apply to
+   */
+  public grantPatch(...grantees: IGrantee[]): void {
+    this.grant(['patch'], grantees);
+  }
+
+  /**
+   * Grant "delete" permission for the grantee.
+   * @param grantees The resource(s) to apply to
+   */
+  public grantDelete(...grantees: IGrantee[]): void {
+    this.grant(['delete'], grantees);
+  }
+
+  /**
+   * Grant "deletecollection" permission for the grantee.
+   * @param grantees The resource(s) to apply to
+   */
+  public grantDeleteCollection(...grantees: IGrantee[]): void {
+    this.grant(['deletecollection'], grantees);
+  }
+
+  /**
+   * Grant "get", "list", and "watch" permissions for the grantee.
+   * @param grantees The resource(s) to apply to
+   */
+  public grantRead(...grantees: IGrantee[]): void {
+    this.grant(['get', 'list', 'watch'], grantees);
+  }
+
+  /**
+   * Grant "get", "list", "watch", "create", "update", "patch", "delete", and
+   * "deletecollection" permissions for the grantee.
+   *
+   * @param grantees The resource(s) to apply to
+   */
+  public grantReadWrite(...grantees: IGrantee[]): void {
+    this.grant(['get', 'list', 'watch', 'create', 'update', 'patch', 'delete', 'deletecollection'], grantees);
+  }
+
+  /**
+   * Grant permission to perform a list of HTTP verbs on the grantee.
+   *
+   * @param grantees The resource(s) to apply to
+   * @see https://kubernetes.io/docs/reference/access-authn-authz/authorization/#determine-the-request-verb
+   */
+  public grant(verbs: string[], grantees: IGrantee[]): void {
+    for (const grantee of grantees) {
+      this.addRule({
+        apiGroups: grantee.apiGroups,
+        resources: grantee.resources,
+        resourceNames: grantee.resourceNames,
+        verbs,
+      });
+    }
+  }
+
+  /**
+   * List of rules included in this rule.
+   *
+   * Returns a copy. To add a rule, use `addRule()`.
+   */
+  public get rules() {
+    return [...this._rules];
+  }
+}
+
+/**
+ * Represents a namespaced role.
+ */
+export interface IRole extends IResource {
+
+}
+
+/**
+ * Options for `Role`.
+ */
+export interface RoleProps extends RoleCommonProps {
+  /**
+   * The namespace of the resources this role's permissions should apply to
+   * (required).
+   */
+  readonly namespace: string;
+}
+
+/**
+ * Role is a namespaced, logical grouping of PolicyRules that can be referenced
+ * as a unit by a RoleBinding.
+ */
+export class Role extends RoleBase implements IRole {
+  /**
+   * @see base.Resource.apiObject
+   */
+  protected readonly apiObject: ApiObject;
+
+  constructor(scope: Construct, id: string, props: RoleProps) {
+    super(scope, id);
+
+    if (props.metadata?.namespace && props.metadata.namespace !== props.namespace) {
+      throw new Error('If `metadata.namespace` is passed as an option, its value must match the value of `namespace`.');
+    }
+
+    const metadata: ApiObjectMetadata = {
+      ...props.metadata,
+      namespace: props.namespace,
+    };
```

Comment:
I've exposed `namespace` as a top-level prop to make it clear that it's a required field and to make it easier to configure. But since namespace is also configurable through `metadata.namespace`, I had to add this error handling logic.

Anyway I'm open to changing this